### PR TITLE
[inference] update examples and engine

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- Update engine name from CaiInferEngine to InferenceEngine. 
- Merge three examples in one example. 
- Update example file name.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/inference/build_smoothquant_weight.py
Comment (ver217) [inline examples/inference/build_smoothquant_weight.py:1]: Line 56-64 are useless

File: examples/inference/run_llama_inference.py [hunk 1]
```diff
@@ -21,18 +20,24 @@ def run_inference(args):
...
     rank = dist.get_rank()
 
-    tokenizer = LlamaTokenizer.from_pretrained(llama_model_path)
-    tokenizer.pad_token_id = tokenizer.unk_token_id
-    model = LlamaForCausalLM.from_pretrained(llama_model_path, pad_token_id=tokenizer.eos_token_id)
-    model = model.half()
+    if args.quant is None:
+        tokenizer = LlamaTokenizer.from_pretrained(llama_model_path)
+        tokenizer.pad_token_id = tokenizer.unk_token_id
+        model = LlamaForCausalLM.from_pretrained(llama_model_path, pad_token_id=tokenizer.eos_token_id)
```
Comment (ver217) [inline examples/inference/run_llama_inference.py:26]: This is strange. You set `pad_token_id=unk_token_id` for tokenzier, but `pad_token_id=eos_token_id` for model

File: examples/inference/run_llama_inference.py [hunk 2]
```diff
@@ -21,18 +20,24 @@ def run_inference(args):
...
         )
-    )
+    elif args.quant == "smoothquant":
+        from colossalai.inference.quant.smoothquant.models.llama import SmoothLlamaForCausalLM
+
+        model = SmoothLlamaForCausalLM.from_quantized(llama_model_path, model_basename=args.smoothquant_base_name)
+        model = model.cuda()
 
-    engine = CaiInferEngine(
+    engine = InferenceEngine(
```
Comment (ver217) [inline examples/inference/run_llama_inference.py:40]: You should pass quant option, right?

File: examples/inference/run_llama_inference.py
Comment (ver217) [inline examples/inference/run_llama_inference.py:1]: As this is an example, we should use real tokenizer and real inputs.

**Review by ver217 — CHANGES_REQUESTED** [review]

File: examples/inference/run_llama_inference.py
Comment (ver217) [reply examples/inference/run_llama_inference.py:1]: By the way, we don't need to record inference time, as this is not benchmark

**Review by ver217 — APPROVED** [review]

File: examples/inference/build_smoothquant_weight.py
Comment (Xu-Kai) [reply examples/inference/build_smoothquant_weight.py:1]: Remove.

File: examples/inference/run_llama_inference.py (hunk 1 above)
Comment (Xu-Kai) [reply examples/inference/run_llama_inference.py]: Updated.

File: examples/inference/run_llama_inference.py (hunk 2 above)
Comment (Xu-Kai) [reply examples/inference/run_llama_inference.py:42]: Updated.

File: examples/inference/run_llama_inference.py
Comment (Xu-Kai) [reply examples/inference/run_llama_inference.py:1]: Remove benchmark. TODO: real tokenizer and real inputs

**Pull request merged**